pilot/pkg/model/push_context.go

```diff
@@ -860,9 +860,18 @@ func (ps *PushContext) GetAllServices() []*Service {
 }
 
 // ServiceForHostname returns the service associated with a given hostname following SidecarScope
+// It returns the service in the proxy namespace in high priority.
```

Comment:
// It returns the service in the proxy namespace in high priority. -> It returns the service in the proxy namespace if available otherwise returns an arbitrary service.

Reply:
SGTM